user/reference/bionic.md

```diff
@@ -104,11 +104,20 @@ The following versions of Docker, version control software and compilers are pre
 
 ## Python support
 
-* Supported Python versions: `2.7`, `3.6` or higher.
-* Pre-installed Python versions: `2.7.15`, `3.6.7`, and `3.7.1`.
-* Python `2.7.15` will be used when no language version is explicitly set.
+Images for all languages have Python 2 and 3 installations provided by the distribution. The Python 2 one has pip preinstalled.
 
-If you're getting errors about PyPy `pypy is not installed; attempting download`, use one of the more recent python versions such as `pypy2.7-6.0` or `pypy3.5-6.0`.
+For `language:`: `c` (`cpp`), `node_js` (`elm`), `go`, `generic`, `java` (`groovy`, `clojure`), `php`, `python`, `ruby` --
+additional Python versions are preinstalled: `2.7.15`, `3.6.7`, and `3.7.1` -- manageable with `pyenv`.
+
+With `language: python`, additionally:
+
+* For the version selected with `python:`, a `virtualenv` is activated.
+* The following versions can be selected. `3.6.7` is the default.
```

Comment:
# with #2523, we can now provide a more comprehensive list. Maybe this allows us to clean up the document a little bit.

@Pezi777 I would like your suggestions. Thanks.

Reply:
instead of mentioning the python versions here, we could just link to user/languages/python.md?

Reply:
Linking there is okay for me. The information in this section is for all `language`s, but this specific table is only for `language:python`, so delegating there is appropriate.

Though I would rather have it moved the other way: in perspective, I'd like to merge pages for all Linux environments together similar to https://www.appveyor.com/docs/linux-images-software/ to reduce duplication. I guess the current list can be generated similar to the table in #2523 but I don't readily see how.

Reply:
Note that that table doesn't specify which exact versions archives with underqualified names represent (see https://github.com/travis-ci/docs-travis-ci-com/pull/2413#issue-295070053 why this is a problem).